src/cython/vapoursynth.pyx

```diff
@@ -643,18 +644,20 @@ cdef class CallbackData(object):
     def for_future(self, object future, object wrap_call=None):
         if wrap_call is None:
             wrap_call = lambda func, *args, **kwargs: func(*args, **kwargs)
-        self.callback = self.handle_future
+        self.callback = functools.partial(CallbackData.handle_future,
```

Comment:
Would it be possible to just rename "handle_future" to "callback"?
Then we would not need to use functools.partial.

Reply:
I don't think so, at least not just a rename.  Scroll a few lines up and you'll notice the callback attribute is not exclusively used to handle just future objects.

Any reason specifically not to use functools.partial?

Reply:
Although now that I think about it, what do you think of this?  (I haven't tested it yet.)

```python
    def for_future(self, object future, object wrap_call=None):
        if wrap_call is None:
            wrap_call = lambda func, *args, **kwargs: func(*args, **kwargs)

        def callback(node, n, result, env=self.env):
            if isinstance(result, Error):
                func = future.set_exception
            else:
                func = future.set_result

            with use_environment(env).use():
                wrap_call(func, result)

        self.callback = callback
```

And then we remove the `self.future` and `self.wrap_cb` attributes, and the `handle_future()` method, from the class definition.

Reply:
Wouldn't it be a good idea to make it a classmethod?

That way it would be made clear that this is an additional constructor to CallbackData.